cmake/CMakeLists.txt

```diff
@@ -142,6 +142,7 @@ option(onnxruntime_FUZZ_TEST "Enable Fuzz testing" OFF)
 option(onnxruntime_ENABLE_NVTX_PROFILE "Enable NVTX profile." OFF)
 option(onnxruntime_ENABLE_MEMORY_PROFILE "Enable memory profile." OFF)
 option(onnxruntime_ENABLE_TRAINING "Enable training functionality." OFF)
+option(onnxruntime_ENABLE_TRAINING_ON_DEVICE "Enable training on device." OFF)
```

Comment:
can this dependency be documented? maybe it would be good to check in the code that onnxruntime_ENABLE_TRAINING is set if onnxruntime_ENABLE_TRAINING_ON_DEVICE is and warn otherwise.